src/Adapter/Form/ChoiceProvider/CombinationIdChoiceProvider.php

```diff
@@ -83,7 +84,10 @@ public function __construct(
     public function getChoices(array $options): array
     {
         $options = $this->resolveOptions($options);
-        $combinationIds = $this->combinationRepository->getCombinationIds(new ProductId($options['product_id']));
+        $combinationIds = $this->combinationRepository->getCombinationIds(
+            new ProductId($options['product_id']),
+            $options['shop_id'] ? ShopConstraint::shop($options['shop_id']) : ShopConstraint::allShops()
```

Comment:
Probably this combinationIdChoices and shop_id choices must dynamically update depending on each other.
Because in All shops context I should be able to apply spec.price for any combination in any shop and those combinations might be different in shops. So contextShop combinations != all shops combinations. So I think I should handle this in a dedicated PR via javascript :thinking: 
wdyt @jolelievre?